### ISSUE TITLE
[RISC-V] Set up a CI pipeline environment in OpenJ9

### ISSUE THREAD

**Issue by ChengJin01** [eclipse-openj9/openj9#12699]
As requested by @0xdaryl ,   we should be able to set up a cross-compilated/emulated RISC-V CI testing environment in OpenJ9 for RISC-V, which is similar to what was set up at OMR for RISC-V at https://github.com/eclipse/omr/issues/4641.   As @janvrany starts to open OpenJ9 PRs for his work, it would be very useful if we could verify the changes at the project. 

Actually we previously had cross-compiling jobs at Adopt but it won't work any wore. For now, we still have cross-compiling jobs internally which seems working to generate the cross-build (we need to figure out what is still left to do there)

The issue is created to check what is still missing, what has been done and what to do next.

FYI: @0xdaryl, @AdamBrousseau, @jdekonin, @tajila, @pshipton, @DanHeidinga

**Comment by jdekonin** [comment]
There is already a system setup for OMR to perform cross-compiles for risc-v.  @janvrany having a big role in getting that operational as I understand.  I am not familiar with how that differs from the Adopt cross-compile environment though.

We should be able to clone out the system that OMR uses and add to the OpenJ9 jenkins without too much trouble.  I *believe* all the pipelines, whether nightly or pr based, should be created automatically when required.

Next actions would include
1. make a deb10 cross-compile node available to openj9 jenkins
2. verify required platform identifiers are committed/updated
- buildenv/jenkins/variables/defaults.yml
- buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy
3. test PR triggers for build and whatever levels of testing are believed to be functional or value in running
4. if successful PR compile add to nightlies

Does this sound about right @AdamBrousseau?  Am I missing anything?

**Comment by AdamBrousseau** [comment]
Adopt cross compiles in a cent6 docker container on x. They do have actual machines they run testing on. Does the omr deb10 do cross compiles or does it do "real" compiles on an emulated environment? I'm guessing the latter. If we do cross compile route, we should be able to use any of our existing cent6 machines or the container once Joe gets it going, maybe with additional tools. But that won't help with test.

Those 2 files are the ones to update for adding it to the pipelines. I doubt Pete will want it in the regular nightly because of the amount of testing that runs there. Maybe it will need its own nightly.

**Comment by janvrany** [comment]
> Does the omr deb10 do cross compiles or does it do "real" compiles on an emulated environment? I'm guessing the latter.

No, the OMR RISC-V job cross-compiles on x86_64 and then uses QEMU in process-level emulation mode to run tests.

**Comment by jdekonin** [comment]
@AdamBrousseau create a PR #13032 to get a pipeline functional.  Question for @ChengJin01 or anyone that can answer...does riscv require freemarker.jar (UMA) or is it dependency for just cmake?

**Comment by ChengJin01** [comment]
Hi @jdekonin,  the cross-compilation via UMA is already supported by the existing changes. But it looks like https://github.com/eclipse-openj9/openj9/pull/13032 is used to configure/compile with `cmake` enabled by default. I suspect `cmake` for RISC-V is not ready for cross-compliation. FYI: @dnakamura
